zeebe/exporters/opensearch-exporter/src/main/java/io/camunda/zeebe/exporter/opensearch/OpensearchMetrics.java

```diff
@@ -7,66 +7,63 @@
-import io.prometheus.client.Gauge;
-import io.prometheus.client.Histogram;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.Timer.ResourceSample;
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicInteger;
 
 public class OpensearchMetrics {
-  private static final String NAMESPACE = "zeebe_opensearch_exporter";
+  private static final String NAMESPACE = "zeebe.opensearch.exporter";
   private static final String PARTITION_LABEL = "partition";
 
-  private static final Histogram FLUSH_DURATION =
-      Histogram.build()
-          .namespace(NAMESPACE)
-          .name("flush_duration_seconds")
-          .help("Flush duration of bulk exporters in seconds")
-          .labelNames(PARTITION_LABEL)
-          .register();
-
-  private static final Counter FAILED_FLUSH =
-      Counter.build()
-          .namespace(NAMESPACE)
-          .name("failed_flush")
-          .help("Number of failed flush operations")
-          .labelNames(PARTITION_LABEL)
-          .register();
-
-  private static final Histogram BULK_SIZE =
-      Histogram.build()
-          .namespace(NAMESPACE)
-          .name("bulk_size")
-          .help("Exporter bulk size")
-          .buckets(10, 100, 1_000, 10_000, 100_000)
-          .labelNames(PARTITION_LABEL)
-          .register();
-
-  private static final Gauge BULK_MEMORY_SIZE =
-      Gauge.build()
-          .namespace(NAMESPACE)
-          .name("bulk_memory_size")
-          .help("Exporter bulk memory size")
-          .labelNames(PARTITION_LABEL)
-          .register();
-
   private final String partitionIdLabel;
+  private final MeterRegistry meterRegistry;
+  private final AtomicInteger bulkMemorySize = new AtomicInteger(0);
 
-  public OpensearchMetrics(final int partitionId) {
+  public OpensearchMetrics(final int partitionId, final MeterRegistry meterRegistry) {
     partitionIdLabel = String.valueOf(partitionId);
+    this.meterRegistry = meterRegistry;
   }
 
-  public Histogram.Timer measureFlushDuration() {
-    return FLUSH_DURATION.labels(partitionIdLabel).startTimer();
+  public ResourceSample measureFlushDuration() {
+    return Timer.resource(meterRegistry, meterName("flush.duration.seconds"))
+        .description("Flush duration of bulk exporters in seconds")
+        .tag(PARTITION_LABEL, partitionIdLabel)
+        .publishPercentileHistogram()
+        .minimumExpectedValue(Duration.ofMillis(10));
   }
 
   public void recordBulkSize(final int bulkSize) {
-    BULK_SIZE.labels(partitionIdLabel).observe(bulkSize);
+    DistributionSummary.builder(meterName("bulk.size"))
+        .description("Exporter bulk size")
+        .tags(PARTITION_LABEL, partitionIdLabel)
+        .serviceLevelObjectives(10, 100, 1_000, 10_000, 100_000)
```

Comment:
:question: should we add more buckets in the 1KB - 32KB range?

Reply:
Yes probably, but not in this PR :smile: